crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -140,6 +136,14 @@ pub(crate) fn unnecessary_regular_expression(checker: &mut Checker, call: &ExprC
     checker.diagnostics.push(diagnostic);
 }
 
+pub(crate) fn string_has_metacharacters(value: &StringLiteralValue) -> bool {
+    // Compare to the complete list from
+    // https://docs.python.org/3/howto/regex.html#matching-characters
+    value
+        .to_str()
+        .contains(['.', '^', '$', '*', '+', '?', '{', '[', '\\', '|', '('])
+}
```

Comment:
it seems like you're no longer using this function from the other rule, so I think all changes to this file are unnecessary for this PR. Could you revert them?